cmd/containerd-shim-runhcs-v1/service_internal.go

```diff
@@ -107,7 +107,7 @@ func (s *service) createInternal(ctx context.Context, req *task.CreateTaskReques
 	// If sandbox isolation is set to hypervisor, make sure the HyperV option
 	// is filled in. This lessens the burden on Containerd to parse our shims
 	// options if we can set this ourselves.
-	if shimOpts.SandboxIsolation == runhcsopts.Options_HYPERVISOR {
+	if shimOpts != nil && shimOpts.SandboxIsolation == runhcsopts.Options_HYPERVISOR {
```

Comment:
Instead of needing to handle `nil` each time we check `shimOpts`, maybe just change the initialization code above so we declare it as `var shimOpts runhcsopts.Options` (no `*`), so then it gets a proper zero value for the struct?

Reply:
Or shimOpts := &runhcs.Options{}

Reply:
Yea that's better